src/lib.rs

```diff
@@ -270,21 +270,18 @@ impl SteamDir {
         // Steam's installation location is pretty easy to find on Linux, too, thanks to the symlink in $USER
         let home_dir = dirs::home_dir()?;
 
-        // Check for Flatpak steam install
-        let steam_flatpak_path = home_dir.join(".var/app/com.valvesoftware.Steam");
-        if steam_flatpak_path.is_dir() {
-            let steam_flatpak_install_path = steam_flatpak_path.join(".steam/steam");
-            if steam_flatpak_install_path.is_dir() {
-                return Some(steam_flatpak_install_path);
-            }
-        }
-
-        // Check for Standard steam install
-        let standard_path = home_dir.join(".steam/steam");
-        if standard_path.is_dir() {
-            return Some(standard_path);
-        }
+        let steam_paths = vec![
+            // Flatpak steam install directories
+            home_dir.join(".var/app/com.valvesoftware.Steam/.local/share/Steam"),
+            home_dir.join(".var/app/com.valvesoftware.Steam/.steam/steam"),
+            home_dir.join(".var/app/com.valvesoftware.Steam/.steam/root"),
+            // Standard install directories
+            home_dir.join(".local/share/Steam"),
+            home_dir.join(".steam/steam"),
+            home_dir.join(".steam/root"),
+            home_dir.join(".steam"),
```

Comment:
AFK, so can't check on my system currently, but when are these different paths used, or more importantly when is `.steam/steam` _not_ valid?

From what I remember `.steam/steam` is a symlink to `.local/share/Steam`, but I would imagine that `.steam/root` and `.steam` would be totally different directories than `SteamDir` is expecting

Reply:
`.steam/root` and `.steam/steam` are generally always symlinks into `.local/share/Steam`, however we've noticed inconsistencies. Sometimes they both exist, sometimes it's either/or.

`.steam` is the result of some odd Ubuntu package management that we experienced a couple years ago. Our tooling checks it just in case, however I admit it's *probably* not applicable for this crate. That path (and many of the others) were added in [this PR](https://github.com/ebkr/r2modmanPlus/pull/398).

Older Debian packages have historically had problems; sometimes choosing `~/Steam`, sometimes, `~/.steam/steam`, sometimes the `.steam` directory itself. It's an absolute *mess*.

Reply:
That all sounds reasonable and the ordering that things are checked seems good 👍